### DISCUSSION TITLE
Align capacitor files structute to allow appflow builds

### DISCUSSION THREAD

**Discussion by devMls** [quasarframework/quasar#16823] [Ideas / Proposals]
Appflow is a common way to build ans publish capacitor apps

Because manage the publication and because compile un the cloud .It help to avoid a mac for compile to ios 

With quasar framework capacitor config is in a subfolder and appflow can't deal with It

I can understand that this IS not  a bug but It is a integration problema bettewn 2 tecnologías that should run together

**Comment by yusufkandemir** [comment]
https://ionic.io/docs/appflow/cookbook/appflow-config
Create `/appflow.config.json` in the root of your project with the following contents:
```json
{
  "apps":
    [
        {
            "appId": "android-app-id",
            "dependencyInstallCommand": "pnpm i",
            "webBuildCommand": "pnpm quasar build -m capacitor -T android",
            "androidPath": "src-capacitor/android",
        },
        {
            "appId": "ios-app-id",
            "dependencyInstallCommand": "pnpm i",
            "webBuildCommand": "pnpm quasar build -m capacitor -T ios",
            "iosPath": "src-capacitor/ios",
        },
    ]
}
```
I have no experience with the tool, I only checked the docs and found it. Quasar does not have to scaffold this file nor it's an integration problem. It's just a matter of configuration. If you confirm this configuration works, we can consider adding it to the documentation so that other people may benefit from it.

**Comment by devMls** [comment]
thanks!!

**Reply by yusufkandemir** [reply]
Please let us know whether it works or not.

**Reply by FelipeSD** [reply]
it didn't worked for me, but I found another config for /appflow.config.json to work with quasar [here](https://stackoverflow.com/questions/66095757/how-can-i-use-ionic-appflow-with-a-quasar-capacitor-project#answer-77224404).

```javascript
{
    "apps":
    [
        {
            "appId": "yourAppflowAppId",
            "root": "src-capacitor",
            "dependencyInstallCommand": "cd .. && npm install",
            "webBuildCommand": "cd .. && npx quasar build -m capacitor -T ios --skip-pkg && npx quasar build -m capacitor -T android --skip-pkg"
        }
    ]
}
```

**Reply by yusufkandemir** [reply]
@FelipeSD can you tell us what didn't work, please? Did it give any errors, warnings, etc.?

**Reply by FelipeSD** [reply]
hello @yusufkandemir, 
Yes, it appears that when using Capacitor, Appflow needs to locate the capacitor.config.json file. When I initially ran the build with the previous configuration, it returned the following error:
<img width="479" alt="image" src="https://github.com/quasarframework/quasar/assets/18513213/034c14e2-0bd4-4431-92a9-0be280419633">
The solution was to define the root directory as src-capacitor, where the capacitor.config.json is located, and then change directories to install dependencies.
